inference_perf/loadgen/load_generator.py

```diff
@@ -218,7 +234,12 @@ async def run_stage(
         # don't miss the initial scheuled request times
         start_time_epoch = time.time()
         start_time = time.perf_counter() + 1
-        num_requests = int(rate * duration)
+
+        if self.datagen.trace is not None:
+            num_requests = self.datagen.get_request_count()
```

Comment:
We do now support timing out a stage, maybe it makes sense for the trace loadgen config to support a duration.